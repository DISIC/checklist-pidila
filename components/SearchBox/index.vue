<template>
  <div class="filter-boxes">
    <h2>Rechercher dans les critères</h2>
    <form @submit.prevent="submit">
      <div class="input-group search-group">
        <input class="form-control" type="search" v-model="queryString"
               title="Rechercher dans l'ensemble des critères (exemple: accueil)" placeholder="ex. : accueil"/>
        <button type="submit" class="btn btn-search">
          <svg class="svg-icon icon-search" aria-hidden="true" focusable="false">
            <use xlink:href="svg/icon-sprite.svg#search"></use>
          </svg>
          <span class="sr-only">Valider la recherche</span>
        </button>
      </div>
    </form>
  </div>
</template>

<script>
  import Events from '~/bus/events'

  export default {
    name: 'SearchBox',
    data: () => {
      return {
        queryString: ''
      }
    },
    methods: {
      submit: function () {
        this.$store.commit('SET_SEARCH_QUERY', this.queryString)
      }
    },
    mounted () {
      this.$bus.$on(Events.STORE_INITIALIZED, () => {
        this.queryString = this.$store.state.searchQuery
      })
    }
  }
</script>
