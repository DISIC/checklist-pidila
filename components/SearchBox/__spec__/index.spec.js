import {shallowMount} from '@vue/test-utils'
import SearchBox from '../index.vue'
import Events from '../../../bus/events'

describe('components | SearchBox', () => {
  let $store = {}
  let $bus = {}
  beforeEach(() => {
    $store.state = {
      searchQuery: ''
    }
    $store.commit = jest.fn()
    $bus.$on = jest.fn()
  })

  it('should match snapshot', () => {
    // when
    const wrapper = shallowMount(SearchBox, {mocks: {$store, $bus}})

    // then
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should contain one input text for searching', () => {
    // when
    const wrapper = shallowMount(SearchBox, {mocks: {$store, $bus}})

    // then
    expect(wrapper.findAll('input')).toHaveLength(1)
  })

  it('should contain one button "Rechercher" to trigger search', () => {
    // when
    const wrapper = shallowMount(SearchBox, {mocks: {$store, $bus}})

    // then
    expect(wrapper.findAll('button')).toHaveLength(1)
  })

  it('should call a mutation SET_SEARCH_QUERY from store when button is clicked', () => {
    // given
    const wrapper = shallowMount(SearchBox, {mocks: {$store, $bus}})

    // when
    wrapper.find('input[type=search]').element.value = 'coréen'
    wrapper.find('input[type=search]').trigger('input')
    wrapper.find('button[type=submit]').trigger('submit')

    // then
    expect($store.commit).toHaveBeenCalledWith('SET_SEARCH_QUERY', 'coréen')
  })

  it('should set input value with value from store', () => {
    // given
    $store.state.searchQuery = 'image'
    let searchBoxInputInitialize
    $bus.$on = function (eventKey, callback) {
      if (eventKey === Events.STORE_INITIALIZED) {
        searchBoxInputInitialize = callback
      }
    }
    const wrapper = shallowMount(SearchBox, {mocks: {$store, $bus}})

    // when
    searchBoxInputInitialize()

    // then
    const inputValue = wrapper.find('input[type=search]').element.value
    expect(inputValue).toBe('image')
  })
})
