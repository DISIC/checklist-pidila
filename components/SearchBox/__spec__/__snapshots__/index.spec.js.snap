// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | SearchBox should match snapshot 1`] = `
<div
  class="filter-boxes"
>
  <h2>
    Rechercher dans les critères
  </h2>
   
  <form>
    <div
      class="input-group search-group"
    >
      <input
        class="form-control"
        placeholder="ex. : accueil"
        title="Rechercher dans l'ensemble des critères (exemple: accueil)"
        type="search"
      />
       
      <button
        class="btn btn-search"
        type="submit"
      >
        <svg
          aria-hidden="true"
          class="svg-icon icon-search"
          focusable="false"
        >
          <use
            xlink:href="svg/icon-sprite.svg#search"
          />
        </svg>
         
        <span
          class="sr-only"
        >
          Valider la recherche
        </span>
      </button>
    </div>
  </form>
</div>
`;
